src/main/java/com/google/sps/model/GmailClientImpl.java

```diff
@@ -125,6 +130,46 @@ public Message getUserMessageWithMetadataHeaders(String messageId, List<String>
         .collect(Collectors.toList());
   }
 
+  @Override
+  public int getWordCount(int numberDaysUnread) {
+    GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.FULL;
+    int wordCount = 0;
+    List<Message> unreadMessages = new ArrayList<>();
+    try {
+      unreadMessages = getUnreadEmailsFromNDays(messageFormat, numberDaysUnread);
+    } catch (IOException e) {
+      System.out.println(e);
+    }
+    for (Message message : unreadMessages) {
+      try {
+        wordCount += getMessageSize(message);
+      } catch (MessagingException | IOException e) {
+        System.out.println(e);
+      }
+    }
```

Comment:
I think something like this is a little bit more cleaner
```
try {
  getUnreadEmailsFromNDays(messageFormat, numberDaysUnread)
    .stream()
    .map(message -> wordCount += getMessageSize(message));
} catch (MessagingException | IOException e) {
  System.out.println(e);
}
```

Reply:
I have now separated the logic to get the List of messages and to do the word count in two different classes. I will try to implement this for earlier PRs.

Reply:
Thanks! `RESOLVED`